src/load/azext_load/data_plane/load_test_run/custom.py

```diff
@@ -0,0 +1,223 @@
+            if test_run_data.get("testArtifacts", {}).get("outputArtifacts", {}).get("resultFileInfo") is not None:
+                url = test_run_data.get("testArtifacts", {}).get("outputArtifacts", {}).get("resultFileInfo").get("url")
+                file_name = test_run_data.get("testArtifacts", {}).get("outputArtifacts", {}).get("resultFileInfo", {}).get("fileName")
+                file_path = os.path.join(path, file_name)
+                response = requests.get(url)
+                with open(file_path, "wb") as f:
+                    f.write(response.content)
+                logger.warning("Results file downloaded to %s", file_path)
+            else:
+                logger.info("No results file found for test run %s", test_run_id)
+        else:
+            logger.info("No results file and output artifacts found for test run %s", test_run_id)
+
+def add_test_run_app_components(
+    cmd,
+    load_test_resource,
+    test_run_id,
+    app_component_id,
+    app_component_name,
+    app_component_type,
+    app_component_kind=None,
+    resource_group_name=None,
+):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    body = {
+        "testRunId": test_run_id,
+        "components": {
+            app_component_id: {
+                "resourceId": app_component_id,
+                "resourceName": app_component_name,
+                "resourceType": app_component_type,
+            }
+        },
+    }
+    if app_component_kind:
+        body["components"][app_component_id]["kind"] = app_component_kind
+    logger.debug("Adding app component to the test run... %s", body)
+    return client.create_or_update_app_components(test_run_id=test_run_id, body=body)
+
+
+def list_test_run_app_components(
+    cmd,
+    load_test_resource,
+    test_run_id,
+    resource_group_name=None,
+):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    logger.debug("Listing app components for the given test run...")
+    return client.get_app_components(test_run_id=test_run_id)
+
+def remove_test_run_app_components(
+    cmd,
+    load_test_resource,
+    test_run_id,
+    app_component_id,
+    resource_group_name=None,
+):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    body = {"testRunId": test_run_id, "components": {app_component_id: None}}
+    logger.debug("Removing app component from the test run... %s", body)
+    return client.create_or_update_app_components(test_run_id=test_run_id, body=body)
+
+def get_client_metrics(cmd, load_test_resource, test_run_id, resource_group_name=None):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+
+    test_run_response = client.get_test_run(test_run_id)
+
+    metric_namespaces = client.get_metric_namespaces(test_run_id)
+
+    metric_definitions = client.get_metric_definitions(
```

Comment:
Where is this used?

Reply:
This is WIP for supporting metrics. Will update